src/hotspot/share/gc/g1/g1HeapRegion.cpp

```diff
@@ -156,6 +157,7 @@ void G1HeapRegion::clear_both_card_tables() {
 void G1HeapRegion::set_free() {
   if (!is_free()) {
     report_region_type_change(G1HeapRegionTraceType::Free);
+    record_activity(); // Record timestamp when region becomes free
```

Comment:
Make this unconditional. At startup, the default value of the region's type is `Free`, so this will not show up if removing the call to `hr_clear()`.